torchgeo/datasets/nccm.py

```diff
@@ -0,0 +1,208 @@
+    smoothed 10-day Sentinel-2 time series data and
+    optimized features from spectral, temporal and
+    textural characteristics of the land surface.
+    The resultant maps have high overall accuracies (OA)
+    based on ground truth data. The dataset contains information
+    specific to three years: 2017, 2018, 2019.
+
+    The dataset contains 5 classes:
+
+    * 0. paddy rice
+    * 1. maize
+    * 2. soybean
+    * 3. others crops and lands
+    * 15. no-data
+
+    Dataset format:
+
+    * Three .TIF files containing the labels
+    * JavaScript code to download images from the dataset.
+
+    If you use this dataset in your research, please cite the following paper:
+
+    * https://doi.org/10.1038/s41597-021-00827-9
+
+    .. versionadded:: 0.6
+    """
+
+    filename_regex = r"CDL(?P<year>\d{4})_clip"
+    filename_glob = "CDL*.*"
+    zipfile_glob = "13090442.zip"
+
+    date_format = "%Y"
+    is_image = False
+    url = "https://figshare.com/ndownloader/articles/13090442/versions/1"
+    md5 = "eae952f1b346d7e649d027e8139a76f5"
+
+    cmap = {
+        0: (0, 255, 0, 255),
+        1: (255, 0, 0, 255),
+        2: (255, 255, 0, 255),
+        3: (128, 128, 128, 255),
+        15: (255, 255, 255, 255),
+    }
+
+    def __init__(
+        self,
+        paths: Union[str, Iterable[str]] = "data",
+        crs: Optional[CRS] = None,
+        res: Optional[float] = None,
+        transforms: Optional[Callable[[dict[str, Any]], dict[str, Any]]] = None,
+        cache: bool = True,
+        download: bool = False,
+        checksum: bool = False,
+    ) -> None:
+        """Initialize a new dataset.
+
+        Args:
+            paths: one or more root directories to search or files to load
+            crs: :term:`coordinate reference system (CRS)` to warp to
+                (defaults to the CRS of the first file found)
+            res: resolution of the dataset in units of CRS
+                (defaults to the resolution of the first file found)
+            transforms: a function/transform that takes an input sample
+                and returns a transformed version
+            cache: if True, cache file handle to speed up repeated sampling
+            download: if True, download dataset and store it in the root directory
+            checksum: if True, check the MD5 after downloading files (may be slow)
+
+        Raises:
+            FileNotFoundError: if no files are found in ``paths``
+            RuntimeError: if ``download=False`` but dataset is missing or checksum fails
+        """
+        self.paths = paths
+        self.download = download
+        self.checksum = checksum
+        self.ordinal_map = torch.zeros(max(self.cmap.keys()) + 1, dtype=self.dtype)
+        self.ordinal_cmap = torch.zeros((5, 4), dtype=torch.uint8)
+
+        self._verify()
+        super().__init__(paths, crs, res, transforms=transforms, cache=cache)
+
+        for v, k in enumerate(list(self.cmap.keys())):
```

Comment:
`v` is kind of confusing, at first I thought it was (value, key) pairs, but it's actually (index, key) pairs. Maybe use `i` instead. You could also use `i, (k, v) in enumerate(self.cmap.items())` if you don't want to index the dict later. Not sure if `list(...)` is needed.